Source/FieldSolver/WarpXPushFieldsHybridPIC.cpp

```diff
@@ -257,34 +230,74 @@ void WarpX::HybridPICEvolveFields ()
     }
 }
 
+void WarpX::HybridPICDepositRhoAndJ ()
```

Comment:
Should this be in the WarpX class? I think this could be abstracted a bit to be a separate class that is friend's with the WarpX class to modularize it a bit. Not a deal breaker though since the main loops in the Field solver are in WarpX currently. It would be a bit of a refactor and could take place in another PR, however this could be a time to discuss architecture a bit.

Reply:
That sounds good as a larger, follow-up refactor to the overall structure.